### ISSUE TITLE
[CI] Verify version constants failing on pre-8.0 clusters with FIPS enabled

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#85640]
This looks to have been failing for awhile. https://gradle-enterprise.elastic.co/s/2zxpfgsfn4s5k/failure

```
:qa:verify-version-constants:v7.17.3#integTest FAILED
Exec output and error:
| Output for ./bin/elasticsearch-users:
| ERROR: Only PBKDF2 is allowed for password hashing in a FIPS 140 JVM. Please set the appropriate value for [ xpack.security.authc.password_hashing.algorithm ] setting.
```

It's possible here this is an issue with _any_ BWC tests against 7.17 clusters as it seems we've changed some stuff related to FIPS between now and then?

**Comment by n1v0lg** [comment]
This failed again here: https://gradle-enterprise.elastic.co/s/lixfwmgr3ebgw

**Comment by n1v0lg** [comment]
I've dug into this some more:

* This issue has been around a while, possibly as far back as Dec 2020 (https://gradle-enterprise.elastic.co/s/h4qgrf6mzd2m2
) 
* We added a check to `UserTools` that throws if fips mode is enabled but the hashing algo is not fips-compliant (#85562) for v8
* We backported above check into 7.x (#55734)
* We added functionality that picks a fips-compliant hashing algo as a default setting if fips is enabled for v8 (before this the default was always non-fips compliant)

`UserTools` is used as part of setting up certain int and bwc tests. It works fine for v8 because we default to the correct hashing algorithm. It breaks for v7 because we do not which fails the validation and consequently test setup.

**tl;dr** this isn't a backwards compatibility break as far as non-test code is concerned but rather a spaghetti situation around how configure settings for fips

I'll look into a fix tomorrow.

**Comment by pugnascotia** [comment]
Another failure: https://gradle-enterprise.elastic.co/s/cnwaf54tl6a7y/console-log?task=:qa:verify-version-constants:v7.17.3%23integTest
